Tests/scripts/lock_cloud_machines.py

```diff
@@ -0,0 +1,359 @@
+from Tests.scripts.utils import logging_wrapper as logging
+from Tests.scripts.utils.log_util import install_logging
+from time import sleep
+import random
+import requests
+from google.cloud import storage
+import argparse
+
+
+QUEUE_REPO = 'queue'
+MACHINES_LOCKS_REPO = 'machines_locks'
+
+
+def options_handler():
+    """
+    Returns: options parsed from input arguments.
+
+    """
+    parser = argparse.ArgumentParser(description='Utility for lock machines')
+    parser.add_argument('--service_account', help='Path to gcloud service account.')
+    parser.add_argument('--gcs_locks_path', help='Path to lock repo.')
+    parser.add_argument('--ci_job_id', help='the job id.')
+    parser.add_argument('--test_machines_path', help='the name of the file with all the test machines.')
+    parser.add_argument('--gitlab_status_token', help='gitlab token to get the job status.')
+    parser.add_argument('--response_machine', help='file to update the chosen machine.')
+    parser.add_argument('--lock_machine_name', help='a machine name to lock the specific machine')
+    parser.add_argument('--number_machines_to_lock', help='the needed machines number.', type=int)
+
+    options = parser.parse_args()
+    return options
+
+
+def get_queue_locks_details(storage_client: storage.Client, bucket_name: str, prefix: str):
+    """
+    get a list of all queue locks files.
+    Args:
+        storage_client(storage.Client): The GCP storage client.
+        bucket_name(str): the bucket name.
+        prefix(str): the prefix to search for specific files.
+
+    Returns: list of dicts with the job-id and the time_created of the lock file.
+
+    """
+    blobs = storage_client.list_blobs(bucket_name)
+    files = []
+    found = False
+    for blob in blobs:
```

Comment:
Are you iterating for all these folders? 
If yes why? 

![image](https://user-images.githubusercontent.com/72339940/222145474-0efae1c0-1eba-45e0-b09f-db82a07c4fce.png)

Reply:
I have not found another way that works in our Becket.
(there is an option to do it through an API call, which I will check)
In our case, our files are at the beginning of the list, so it doesn't actually go through the entire bucket (but it's obviously not the best way)

Reply:
Why not putting both **content-locks-xsiam** and **content-locks-xsoar-ng** in **content-locks**?
also when we are done we need to delete the test folder.

Reply:
It will be easier to debug if we will have different folders.

Reply:
I did not say not to have different folders, I said to put them both under **content-locks**. It should be separated from other artifacts (tests, content, etc...)